tools/sambamba.wdl

```diff
@@ -0,0 +1,289 @@
+        queryname_sort: {
+            description: "If true, sort the BAM by queryname. If false, sort by coordinate.",
+            common: true
+        }
+        modify_disk_size_gb: "Add to or subtract from dynamic disk space allocation. Default disk size is determined by the size of the inputs. Specified in GB."
+        ncpu: "Number of cores to allocate for task"
+    }
+
+    input {
+        File bam
+        String prefix = basename(bam, ".bam") + ".sorted"
+        Boolean queryname_sort = false
+        Int modify_disk_size_gb = 0
+        Int ncpu = 2
+    }
+
+    Float bam_size = size(bam, "GiB")
+    Int disk_size_gb = ceil(bam_size * 3) + 10 + modify_disk_size_gb
+
+    String outfile_name = prefix + ".bam"
+
+    command <<<
+        set -euo pipefail
+
+        sambamba sort \
+            --nthreads ~{ncpu} \
+            -o ~{outfile_name} \
+            ~{if queryname_sort then '-n' else ''} \
+            ~{bam}
+    >>>
+
+    output {
+        File sorted_bam = outfile_name
+    }
+
+    runtime {
+        cpu: ncpu
+        memory: "25 GB"
+        disk: "~{disk_size_gb} GB"
+        container: 'quay.io/biocontainers/sambamba:1.0--h98b6b92_0'
+        maxRetries: 1
+    }
+}
+
+task markdup {
+    meta {
+        description: "Marks duplicate reads in the input BAM file"
+        outputs: {
+            duplicate_marked_bam: "The input BAM with computationally determined duplicates marked.",
+            mark_duplicates_metrics: "Duplicate marking metrics output from sambamba"
+        }
+    }
+
+    parameter_meta {
+        bam: "Input BAM format file in which to mark duplicates"
+        prefix: "Prefix for the markdup result files. The extensions `markdup.bam` will be added."
+        remove_duplicates: {
+            description: "If true, remove duplicates instead of marking them.",
+            common: true
+        }
+        modify_disk_size_gb: "Add to or subtract from dynamic disk space allocation. Default disk size is determined by the size of the inputs. Specified in GB."
+        ncpu: "Number of cores to allocate for task"
+    }
+
+    input {
+        File bam
+        String prefix = basename(bam, ".bam")
+        Boolean remove_duplicates = false
+        Int modify_disk_size_gb = 0
+        Int ncpu = 2
+    }
+
+    Float bam_size = size(bam, "GiB")
+    Int disk_size_gb = ceil((bam_size * 2) + 10) + modify_disk_size_gb
+
+    command <<<
+        sambamba markdup \
+            --nthreads ~{ncpu} \
+            ~{if remove_duplicates then '--remove-duplicates' else ''} \
+            ~{bam} \
+            ~{prefix}.markdup.bam \
+            > ~{prefix}.markdup_log.txt
+    >>>
+
+    output {
+        File duplicate_marked_bam = "~{prefix}.markdup.bam"
+        File duplicate_marked_bam_index = "~{prefix}.markdup.bam.bai"
+        File markdup_log = "~{prefix}.markdup_log.txt"
+    }
+
+    runtime {
+        cpu: ncpu
+        memory: "50 GB"
```

Comment:
This should probably be made dynamic. I don't want to hold up the PR with a series of tests to find the right formula though. So really just making a comment, not expecting any change right now.

Reply:
```External sort is not implemented. Thus, memory consumption grows by 2Gb per each 100M reads. Check that you have enough RAM before running the tool.```
I'm curious if that is actually proportional to the read count or if it is actually proportional to the size of the file.

Reply:
Or, the more likely third option: neither lol.
Which is to say I imagine it correlates to both, and tracks neither perfectly.